src/js/common/components/RequiredLoginView.jsx

```diff
@@ -65,23 +65,21 @@ class RequiredLoginView extends React.Component {
           return <SystemDownView messageLine1="We couldn’t find your records with that information." messageLine2="Please call the Vets.gov Help Desk at 1-855-574-7286. We’re open Monday‒Friday, 8:00 a.m.‒8:00 p.m. (ET)."/>;
         }
 
-        // If va_profile has any other value, continue on to check if this user can
-        // use this specific service.
-        if (this.isServiceAvailable()) {
-          // If you have the required service, show the application view.
-          return this.props.children;
-        }
-
+        // If va_profile has any other value, show the application view.
         // If the required service is not available, the component will still be rendered,
         // but we pass an `isDataAvailable` prop to child components indicating there is
-        // no data. (Only add this prop to React components (functions), and not ordinary
+        // presence or absence of data. (Only add this prop to React components (functions), and not ordinary
         // DOM elements.)
-        return React.Children.map(this.props.children, (child) => {
-          const props = typeof child.type === 'function'
-            ? { isDataAvailable: false }
-            : null;
-          return React.cloneElement(child, props);
-        });
+        return React.Children.map(this.props.children,
+          (child) => {
+            let props = null;
+            if (typeof child.type === 'function') {
+              props = { isDataAvailable: this.isServiceAvailable() };
+            }
```

Comment:
Also, rather than recreate `props = { isDataAvailable: this.isServiceAvailable() };` for each child, it might be better to create it once outside of the `map`. Although, for the most part, it's usually only going to occur once for a particular child.

```
const props = { isDataAvailable: this.isServiceAvailable() };
return React.Children.map(
  this.props.children,
  (child) => React.cloneElement(child, (typeof child.type === 'function' ? props : null))
);

Reply:
Yeah, I like the above to create the props outside of the `map` and just add it when necessary, but probably a  negligible efficiency gain, so your call.